dart/lib/src/default_integrations.dart

```diff
@@ -7,16 +7,18 @@ import 'sentry.dart';
 import 'sentry_options.dart';
 import 'throwable_mechanism.dart';
 
-/// integration that capture errors on the runZonedGuarded error handler
+/// Integration that runs runner function within runZonedGuarded
 class RunZonedGuardedIntegration extends Integration {
-  final AppRunner _appRunner;
+  RunZonedGuardedIntegration();
 
-  RunZonedGuardedIntegration(this._appRunner);
+  FutureOr<void> Function() runner;
 
   @override
   FutureOr<void> call(Hub hub, SentryOptions options) {
     runZonedGuarded(() async {
-      await _appRunner();
+      if (runner != null) {
+        await runner();
+      }
```

Comment:
if `runner` is null, maybe the integration should not be even executed?
or at least, not even calling `runZonedGuarded(...)` if `runner` is null, we spare a method call, as it does nothing more

Reply:
Now it's rewquired through the constructor.